server/src/main/java/io/druid/server/coordinator/helper/DruidCoordinatorBalancer.java

```diff
@@ -91,13 +91,19 @@ public DruidCoordinatorRuntimeParams run(DruidCoordinatorRuntimeParams params)
         params.getDruidCluster().getCluster().entrySet()) {
       String tier = entry.getKey();
 
-      if (currentlyMovingSegments.get(tier) == null) {
-        currentlyMovingSegments.put(tier, new ConcurrentHashMap<String, BalancerSegmentHolder>());
-      }
+      currentlyMovingSegments.putIfAbsent(tier, new ConcurrentHashMap<>());
```

Comment:
this creates a new `ConcurrentHashMap` regardless of if it is present or not, can you change it to only create a new hash map if absent?